library/std/src/sys/unix/locks/futex_rwlock.rs

```diff
@@ -95,7 +95,11 @@ impl RwLock {
 
         // It's impossible for a reader to be waiting on a read-locked RwLock,
         // except if there is also a writer waiting.
-        debug_assert!(!has_readers_waiting(state) || has_writers_waiting(state));
+        assert!(
```

Comment:
I'm not sure the changes to this file are good. These are impossible states -- they are also pretty hot functions (and they won't necessarily be covered by perf, given that rustc is largely single threaded) which are marked as `#[inline]`, The formatting here will end up with a bunch of additional code in the bodies from the formatting, which will often result in them not getting inlined.

So, I think this should stick to cases where it's checking the result of a syscall. That seems much more justifiable, since those cases are likely not going to be nearly as hot anyway.